### PR TITLE
CI: Add workflow for tagging essential modified files

### PR DESCRIPTION
**Pull request by zvecr**
## Description

Further migration of workflow away from travis.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: .github/workflows/auto_tag.yaml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+name: Essential files modified
+
+on:
+  push:
+    branches:
+    - master
+    paths-ignore:
```
Comment (Erovia) [inline .github/workflows/auto_tag.yaml:7]: What do you think about explicitly including paths/files we consider essential instead of including these dirs?

Comment (zvecr) [reply .github/workflows/auto_tag.yaml:7]: Good point, i will align to the labeler config.

**Review by Erovia — APPROVED** [review]

File: .github/workflows/auto_tag.yaml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
+
+    - name: Bump version and push tag
+      uses: anothrNick/github-tag-action@1.26.0
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        DEFAULT_BUMP: 'none'
```
Comment (skullydazed) [inline .github/workflows/auto_tag.yaml:33]: Should this be `minor` or omitted? Seems like with this change no tagging will happen at all unless we manually `#minor` everything.

Comment (zvecr) [reply .github/workflows/auto_tag.yaml:33]: Was trying to be smart and disable users being able to bump, but i had read the docs wrong
```suggestion
```

**Review by tzarc — APPROVED** [review]

**Pull request merged**